vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/validation/ComboBoxBasicValidationIT.java

```diff
@@ -0,0 +1,106 @@
+package com.vaadin.flow.component.combobox.test.validation;
+
+import com.vaadin.flow.component.combobox.testbench.ComboBoxElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.tests.validation.AbstractValidationIT;
+import org.junit.Test;
+import org.openqa.selenium.Keys;
+
+import static com.vaadin.flow.component.combobox.test.validation.ComboBoxBasicValidationPage.ENABLE_CUSTOM_VALUE_BUTTON;
+import static com.vaadin.flow.component.combobox.test.validation.ComboBoxBasicValidationPage.REQUIRED_BUTTON;
+
+@TestPath("vaadin-combo-box/validation/basic")
+public class ComboBoxBasicValidationIT
+        extends AbstractValidationIT<ComboBoxElement> {
+
+    @Test
+    public void fieldIsInitiallyValid() {
+        assertClientValid();
+        assertServerValid();
+    }
+
+    @Test
+    public void triggerBlur_assertValidity() {
+        testField.sendKeys(Keys.TAB);
+        assertServerValid();
+        assertClientValid();
+    }
+
+    @Test
+    public void required_triggerBlur_assertValidity() {
+        $("button").id(REQUIRED_BUTTON).click();
+
+        testField.sendKeys(Keys.TAB);
+        assertServerInvalid();
+        assertClientInvalid();
+    }
+
+    @Test
+    public void required_changeValue_assertValidity() {
+        $("button").id(REQUIRED_BUTTON).click();
+
+        testField.selectByText("foo");
+        assertServerValid();
+        assertClientValid();
+
+        testField.clear();
+        assertServerInvalid();
+        assertClientInvalid();
+
+        // Try enter custom value
+        testField.sendKeys("custom", Keys.TAB);
+        assertServerInvalid();
+        assertClientInvalid();
+    }
+
+    @Test
+    public void required_customValuesAllowed_assertValidity() {
```

Comment:
Added additional test where custom values are allowed. Probably doesn't make a difference to selecting a predefined value, as the test setup handling custom values is also just changing the component's value. Still can't hurt I guess?